streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopicConfigTest.java

```diff
@@ -16,113 +16,67 @@
 import org.apache.kafka.common.errors.InvalidTopicException;
-import org.apache.kafka.common.utils.Utils;
+import org.apache.kafka.streams.processor.internals.InternalTopicConfig.InternalTopicType;
 import org.junit.Test;
 
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
-import java.util.Properties;
 
 import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertNull;
-import static org.junit.Assert.assertTrue;
 
 public class InternalTopicConfigTest {
 
-    @Test
-    public void shouldHaveCompactionPropSetIfSupplied() {
-        final Properties properties = new InternalTopicConfig("name",
-                                                              Collections.singleton(InternalTopicConfig.CleanupPolicy.compact),
-                                                              Collections.<String, String>emptyMap()).toProperties(0);
-        assertEquals("compact", properties.getProperty(InternalTopicManager.CLEANUP_POLICY_PROP));
-    }
-
-
     @Test(expected = NullPointerException.class)
     public void shouldThrowIfNameIsNull() {
-        new InternalTopicConfig(null, Collections.singleton(InternalTopicConfig.CleanupPolicy.compact), Collections.<String, String>emptyMap());
+        new InternalTopicConfig(null, InternalTopicType.REPARTITION, Collections.<String, String>emptyMap());
     }
 
     @Test(expected = InvalidTopicException.class)
     public void shouldThrowIfNameIsInvalid() {
-        new InternalTopicConfig("foo bar baz", Collections.singleton(InternalTopicConfig.CleanupPolicy.compact), Collections.<String, String>emptyMap());
+        new InternalTopicConfig("foo bar baz", InternalTopicType.REPARTITION, Collections.<String, String>emptyMap());
     }
 
     @Test
-    public void shouldConfigureRetentionMsWithAdditionalRetentionWhenCompactAndDelete() {
+    public void shouldAugmentRetentionMsWithWindowedChangelog() {
         final InternalTopicConfig topicConfig = new InternalTopicConfig("name",
-                                                                        Utils.mkSet(InternalTopicConfig.CleanupPolicy.compact, InternalTopicConfig.CleanupPolicy.delete),
+                                                                        InternalTopicType.WINDOWED_STORE_CHANGELOG,
                                                                         Collections.<String, String>emptyMap());
-        final int additionalRetentionMs = 20;
         topicConfig.setRetentionMs(10);
-        final Properties properties = topicConfig.toProperties(additionalRetentionMs);
-        assertEquals("30", properties.getProperty(InternalTopicManager.RETENTION_MS));
+        assertEquals("30", topicConfig.toProperties(20).get(TopicConfig.RETENTION_MS_CONFIG));
     }
 
-    @Test
-    public void shouldNotConfigureRetentionMsWhenCompact() {
+    @Test(expected = IllegalStateException.class)
```

Comment:
should we apply a try-catch patter instead of annotation? It's not a single line test? (same below)

Reply:
ack.